cache/network/download.go

```diff
@@ -38,44 +43,56 @@ func Download(ctx context.Context, params DownloadParams, logger log.Logger) (ma
 		return "", fmt.Errorf("cache key list is empty")
 	}
 
-	retryableHTTPClient := prepareRetryableHTTPClient(logger)
+	retryableHTTPClient := retryhttp.NewClient(logger)
+	retryableHTTPClient.CheckRetry = customRetryFunction
 	client := newAPIClient(retryableHTTPClient, params.APIBaseURL, params.Token, logger)
 
 	logger.Debugf("Get download URL")
 	restoreResponse, err := client.restore(params.CacheKeys)
 	if err != nil {
 		return "", fmt.Errorf("failed to get download URL: %w", err)
 	}
+
 	logger.Debugf("Download archive")
-	downloadErr := downloadFile(ctx, retryableHTTPClient.StandardClient(), restoreResponse.URL, params.DownloadPath)
-	if downloadErr == nil {
-		return restoreResponse.MatchedKey, nil
+	err = downloadFile(ctx, retryableHTTPClient.StandardClient(), restoreResponse.URL, params.DownloadPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to download archive: %w", err)
 	}
-	return "", fmt.Errorf("failed to download archive")
-}
 
-func prepareRetryableHTTPClient(logger log.Logger) *retryablehttp.Client {
-	retryableHTTPClient := retryhttp.NewClient(logger)
-	retryableHTTPClient.CheckRetry = createCustomRetryFunction()
-	return retryableHTTPClient
+	return restoreResponse.MatchedKey, nil
 }
 
-func createCustomRetryFunction() func(ctx context.Context, resp *http.Response, err error) (bool, error) {
-	retriableErrors := []string{"Range request returned invalid Content-Length", "EOF", "connection reset"}
-
-	return func(ctx context.Context, resp *http.Response, downloadErr error) (bool, error) {
-		defaultCheckRetry := retryablehttp.DefaultRetryPolicy
-		retry, err := defaultCheckRetry(ctx, resp, downloadErr)
+// customRetryFunction - implements CheckRetry
+//
+// CheckRetry specifies a policy for handling retries. It is called
+// following each request with the response and error values returned by
+// the http.Client. If CheckRetry returns false, the Client stops retrying
+// and returns the response to the caller. If CheckRetry returns an error,
+// that error value is returned in lieu of the error from the request. The
+// Client will close any response body when retrying, but if the retry is
+// aborted it is up to the CheckRetry callback to properly close any
+// response body before returning.
+func customRetryFunction(ctx context.Context, resp *http.Response, downloadErr error) (bool, error) {
+	// First call the DefaultRetryPolicy
+	retry, err := retryablehttp.DefaultRetryPolicy(ctx, resp, downloadErr)
```

Comment:
How about naming this `err` something more descriptive, like `defaultRetryErr`. I found this a bit confusing when reviewing #92.